### DISCUSSION TITLE
Asserting mulwad and mulwadup results to test the behaviour of mulwadup!

### DISCUSSION THREAD

**Discussion by bala-Blockchian** [Cyfrin/assembly-evm-opcodes-and-formal-verification-course#3] [Q&A]
Why are we asserting mulwad and mulwadup results to test the behaviour of mulwadup in video 37 of mathmaster.

**Comment by cromewar** [comment]
Hello @bala-Blockchian, thanks for posting this question, please take a look at this if you will to re-formulate this [question](https://github.com/Cyfrin/assembly-evm-opcodes-and-formal-verification-course/blob/main/how-to-ask-a-question.md)

cc @PatrickAlphaC

**Reply by PatrickAlphaC** [reply]
Good question! We are just doing that to show how Certora finds the edge case we need. And the unit test shows that the edge case Certora found is correct.

**Reply by EngrPips** [reply]
Interesting
